app/src/ui/history/file-list.tsx

```diff
@@ -1,14 +1,48 @@
 import * as React from 'react'
-import { FileChange, mapStatus, iconForStatus } from '../../models/status'
+import * as Path from 'path'
+
+import {
+  AppFileStatus,
+  FileChange,
+  mapStatus,
+  iconForStatus,
+} from '../../models/status'
 import { PathLabel } from '../lib/path-label'
 import { Octicon } from '../octicons'
 import { List } from '../lib/list'
+import { showContextualMenu } from '../main-process-proxy'
+import { IMenuItem } from '../../lib/menu-item'
+
+const RestrictedFileExtensions = ['.cmd', '.exe', '.bat', '.sh']
```

Comment:
Seeing the same copy-pasted code makes me think is a candidate for refactoring into a shared module that's responsible for launching the context menu, but we pass through a number of props like `onOpenInExternalEditor` - which depend on the `AppStore` but end up being proxied to the main process - and I need to think about ways to reorganize things.

Reply:
So, it's not related to me? I don't think that can made these changes.